Tests/HttpPipelineTests/__Snapshots__/HttpPipelineTests/testBasicAuth_Authorized.1.Conn.txt

```diff
@@ -0,0 +1,22 @@
+Step:
```

Comment:
if we moved away from the `Snapshot` protocol and towards the struct version we could create snapshots of middleware where you provide a middleware and starting connection, and then we just apply the middleware and snapshot both connections. would be fun!